hermes/bin/src/packaging/wasm_module.rs

```diff
@@ -0,0 +1,115 @@
+//! Wasm module package.
+
+use std::path::Path;
+
+use super::{
+    copy_dir_recursively_to_package, copy_file_from_dir_to_package, DirNotFoundError,
+    FileNotFoundError,
+};
+use crate::errors::Errors;
+
+/// Wasm module package.
+#[derive(Debug)]
+pub(crate) struct WasmModulePackage {
+    /// hdf5 package instance
+    _package: hdf5::File,
+}
+
+impl WasmModulePackage {
+    /// Config JSON file name.
+    const CONFIG_JSON: &'static str = "config.json";
+    /// Config schema JSON file name.
+    const CONFIG_SCHEMA_JSON: &'static str = "config.schema.json";
+    /// Metadata JSON file name.
+    const METADATA_JSON: &'static str = "metadata.json";
+    /// Module wasm file name.
+    const MODULE_WASM: &'static str = "module.wasm";
+    /// Settings schema JSON file name.
+    const SETTINGS_SCHEMA_JSON: &'static str = "settings.schema.json";
+    /// Share directory name.
+    const SHARE: &'static str = "share";
+
+    /// Create a new Wasm module package from a directory.
+    pub(crate) fn from_dir<P: AsRef<Path>>(path: P) -> anyhow::Result<Self> {
+        let path = path.as_ref();
+        let mut errors = Errors::new();
+
+        let package_path = path.join("module.hdf5");
+        let package = hdf5::File::create(&package_path)?;
+
```

Comment:
I wan't to see blosc compression set up properly and used on this file, i do not want it "left for later", because its the sort of thing that always gets delayed and then never done.  Its much easier to get it working with this simple builder than to try and wedge in and test later.  We have no reason to not use compressed images.  The necessary build options for HDF5 are added above, so it should work OK.

Reply:
I'm going to do it in the separate PR.
Next one